File: src/main/java/org/hwabeag/playerfly/expansions/PlayerFlyExpansion.java
package org.hwabeag.playerfly.expansions;

import java.util.Objects;
import me.clip.placeholderapi.expansion.PlaceholderExpansion;
import org.bukkit.OfflinePlayer;
import org.bukkit.configuration.file.FileConfiguration;
import org.hwabeag.playerfly.PlayerFly;
import org.hwabeag.playerfly.config.ConfigManager;
import org.jetbrains.annotations.NotNull;

public class PlayerFlyExpansion extends PlaceholderExpansion {
    FileConfiguration PlayerFlyConfig = ConfigManager.getConfig("playerfly");
    private final PlayerFly plugin;

    public PlayerFlyExpansion(PlayerFly plugin) {
        this.plugin = plugin;
    }

    public boolean persist() {
        return true;
    }

    public boolean canRegister() {
        return true;
    }

    public @NotNull String getAuthor() {
        return this.plugin.getDescription().getAuthors().toString();
    }

    public @NotNull String getIdentifier() {
        return "playerfly";
    }

    public @NotNull String getVersion() {
        return this.plugin.getDescription().getVersion();
    }

    public String onRequest(OfflinePlayer player, @NotNull String params) {
        if (params.equalsIgnoreCase("get")) { // %playerfly_get%
            String name = ((OfflinePlayer)Objects.requireNonNull(player)).getName();
            return this.PlayerFlyConfig.getString("플라이." + name + ".남은시간") == null ? null : this.PlayerFlyConfig.getString("플라이." + name + ".남은시간");
        } else {
            return null;
        }
    }
}

File: src/main/kotlin/org/hwabeag/playerfly/expansions/PlayerFlyExpansion.kt
package org.hwabeag.playerfly.expansions

import me.clip.placeholderapi.expansion.PlaceholderExpansion
import org.bukkit.OfflinePlayer
import org.hwabeag.playerfly.PlayerFlyPlugin
import org.hwabeag.playerfly.database.DatabaseManager

class PlayerFlyExpansion(
    private val plugin: PlayerFlyPlugin,
    private val databaseManager: DatabaseManager
) : PlaceholderExpansion() {
    override fun persist(): Boolean = true

    override fun canRegister(): Boolean = true

    override fun getAuthor(): String = plugin.description.authors.joinToString(",")

    override fun getIdentifier(): String = "playerfly"

    override fun getVersion(): String = plugin.description.version

    override fun onRequest(player: OfflinePlayer?, params: String): String? {
        if (player == null || !params.equals("get", ignoreCase = true)) {
            return null
        }
        val name = player.name ?: return "0"
        return databaseManager.getData(player.uniqueId, name).minutes.toString()
    }
}
